src/dos/cdrom_image.cpp

```diff
@@ -182,6 +188,57 @@ bool CDROM_Interface_Image::AudioFile::seek(Bit32u offset)
 	return result;
 }
 
+bool CDROM_Interface_Image::AudioFile::read(uint8_t *buffer,
+                                            const uint32_t offset,
+                                            const uint32_t requested_bytes)
+{
+	// Check for logic bugs
+	assertm(buffer != nullptr, "buffer needs to be allocated but is the nullptr [Bug]");
+	assertm(sample != nullptr, "Audio sample needs to be valid, but is the nullptr [Bug]");
+
+	// Guard again valid but no-op case
+	if (requested_bytes == 0)
+		return true;
+
+	// We support DAE from 16-bit, stereo, 44 kHz tracks. If the track doesn't conform to
+	// this, then inform the user. Also, we allow up to 10 DAE-attempts before informing
+	// the user - because some CD Player software will query this interface before using
+	// CDROM-directed playback (not DAE), therefore we don't want to fail in those cases.
+	if (getRate() != REDBOOK_PCM_FRAMES_PER_SECOND || getChannels() != REDBOOK_CHANNELS) {
+		static uint8_t dae_attempts = 0;
+		if (dae_attempts++ > 10) {
+			E_Exit("\n"
+			       "CDROM: Digital Audio Extration (DAE) was attempted with a %s %u kHz\n"
+			       "       track, but DAE is currently only compatible with stereo %u kHz\n"
+				   "       tracks.",
```

Comment:
Sneaky tab hid in here.

Reply:
Got it! Thanks.

Recently switched to 'vscode', and it pattern-matches the indent off the first character in the line; so it auto-tabs everything in, meaning I have to manually correct almost every line.  Even with white-space showing, it's quite subtle.
![2020-02-18_07-40](https://user-images.githubusercontent.com/1557255/74751315-ea81e080-5221-11ea-8235-eadca92b3b17.png)